src/main/UserActivityMonitor.js

```diff
@@ -77,51 +65,52 @@ export default class UserActivityMonitor extends EventEmitter {
       } catch (err) {
         console.log('Error getting system idle time:', err);
       }
-    }, this.config.internalUpdateFrequencyMs);
+    }, this.config.updateFrequencyMs);
   }
 
   /**
    * Stop monitoring system events and idle time
    */
   stopMonitoring() {
-    electron.powerMonitor.off('suspend', this.handleSystemGoingAway);
-    electron.powerMonitor.off('resume', this.handleSystemComingBack);
-    electron.powerMonitor.off('lock-screen', this.handleSystemGoingAway);
-    electron.powerMonitor.off('unlock-screen', this.handleSystemComingBack);
-
     clearInterval(this.systemIdleTimeIntervalID);
   }
 
   /**
-   * Updates internal idle time properties and conditionally triggers updates to user activity status
+   * Updates internal idle time and sets internal user activity state
    *
    * @param {integer} idleTime
    * @private
    */
   updateIdleTime(idleTime) {
     this.idleTime = idleTime;
-
-    if (this.idleTime > this.config.activityTimeoutSec) {
-      this.updateUserActivityStatus(false);
-    } else if (!this.forceInactive && this.idleTime < this.config.activityTimeoutSec) {
-      this.updateUserActivityStatus(true);
+    if (idleTime > this.config.inactiveThresholdSec) {
+      this.setActivityState(false);
+    } else {
+      this.setActivityState(true);
     }
   }
 
   /**
-   * Updates user activity status if changed and triggers a status update
+   * Updates user active state and conditionally triggers a status update
    *
    * @param {boolean} isActive
    * @param {boolean} isSystemEvent – indicates whether the update was triggered by a system event (log in/out, screesaver on/off etc)
    * @private
    */
-  updateUserActivityStatus(isActive = false, isSystemEvent = false) {
+  setActivityState(isActive = false, isSystemEvent = false) {
+    this.isActive = isActive;
+
+    if (isSystemEvent) {
+      this.sendStatusUpdate(true);
+    }
+
     const now = Date.now();
-    if (isActive !== this.isActive) {
-      this.isActive = isActive;
-      this.sendStatusUpdate(now, isSystemEvent);
-    } else if (now - this.lastStatusUpdate > this.config.statusUpdateThresholdMs) {
-      this.sendStatusUpdate(now, isSystemEvent);
+
+    if (isActive && (this.lastSetActive == null || now - this.lastSetActive >= this.config.statusUpdateThresholdMs)) {
+      this.sendStatusUpdate(false);
+      this.lastSetActive = now;
+    } else if (!isActive) {
+      this.lastSetActive = null;
```

Comment:
shouldn't it keep at whatever value was before? otherwise it'll set as inactive next time.

Reply:
If you mean `this.lastSetActive = null` then no, that's being done on purpose to allow for an immediate update when the desktop app detects the user going from inactive to active. Basically allows the desktop app to set you back to online immediately when you start interacting with your computer after being away.